Daemon/Daemon/Rules.m

```diff
@@ -221,8 +221,8 @@ -(Rule*)find:(Event*)event
         //check all (process's) rules
         // same startup item and same path
         // note: * is a wildcard, meaning any match
-        for(Rule* rule in self.rules[key][KEY_RULES])
-        {
+        NSArray<Rule *> *rules = self.rules[key][KEY_RULES];
+        NSUInteger idx = [rules indexOfObjectWithOptions:NSEnumerationConcurrent passingTest:^BOOL(Rule * _Nonnull rule, NSUInteger idx, BOOL * _Nonnull stop) {
```

Comment:
Lookup of the rules is also moved to concurrent perform. 

Might work, might not work. For your consideration.